crates/miden-objects/src/block/account_tree.rs

```diff
@@ -43,7 +241,14 @@ impl AccountTree {
 
     /// Creates a new, empty account tree.
     pub fn new() -> Self {
-        AccountTree { smt: Smt::new() }
+        AccountTree { smt: S::new() }
+    }
+
+    /// Creates a new account tree, based on the given `Smt`.
+    ///
+    /// Particularly relevant for additional initialization required, i.e.
+    pub fn new_from(smt: S) -> Self {
+        AccountTree { smt }
```

Comment:
Related to the above comment: I'd probably remove the current `new()` in favor of implementing `Default` and then rename `new_from()` to just `new()` - i.e., it would be:
```rust
pub fn new(smt: S) -> Self {
    AccountTree { smt }
}
```

Also, the comment on line 249 doesn't seem complete.